### ISSUE TITLE
comma-style fix breaks sparse arrays with nested parens

### ISSUE THREAD

**Issue by rincedd** [eslint/eslint#10273]
**Tell us about your environment**

* **ESLint Version:** 4.19.1
* **Node Version:** 9.11.1
* **npm Version:** 5.8.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
env:
  es6: true
rules:
  comma-style:
    - error
    - last

```
</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
class Thing {}
function f() {}

const testArray = [
  [1],
  ,
  [new Thing(2)],
  [new Thing(3)],
  ,
  [new Thing(4)],
  [new Thing(5)],
  [f(6)],
  ,
  [{an: 'object'}],
  ,
  [[7, 8]],
  ,
  [9]
];

```

```bash
eslint --fix .
```

**What did you expect to happen?**
Comma-style rule should fix the wrong linebreaks before and after the single commas by putting them at the end of the preceding line.

**What actually happened? Please include the actual, raw output from ESLint.**
The comma-style fix places the comma inside of the preceding array literal if this array contains something that ends with a parenthesis ")". This means that the sparse array is no longer sparse at this index, because there are no longer two consecutive commas at that point.

Please note that this does not happen if the preceding array does not end with something that ends in a closing parenthesis. In these cases, the comma is placed at the end of the line correctly, so that there are two consecutive commas at the end of the line after the fix.

```js
class Thing {}
function f() {}

const testArray = [
  [1],,  
  [new Thing(2)],
  [new Thing(3),],  
  [new Thing(4)],
  [new Thing(5)],
  [f(6),],  
  [{an: 'object'}],,  
  [[7, 8]],,  
  [9]
];

```

**Comment by Aladdin-ADD** [comment]
thanks for the report! I can repro it.

**Comment by malcolmsgroves** [comment]
I took a stab at this, seems to be fixed
